src/cdk/drag-drop/directives/drag.ts

```diff
@@ -0,0 +1,284 @@
+  @ContentChild(CdkDragPreview) _previewTemplate: CdkDragPreview;
+
+  /** Template for placeholder element rendered to show where a draggable would be dropped. */
+  @ContentChild(CdkDragPlaceholder) _placeholderTemplate: CdkDragPlaceholder;
+
+  /** Arbitrary data to attach to this drag instance. */
+  @Input('cdkDragData') data: T;
+
+  /** Locks the position of the dragged element along the specified axis. */
+  @Input('cdkDragLockAxis') lockAxis: 'x' | 'y';
+
+  /**
+   * Selector that will be used to determine the root draggable element, starting from
+   * the `cdkDrag` element and going up the DOM. Passing an alternate root element is useful
+   * when trying to enable dragging on an element that you might not have access to.
+   */
+  @Input('cdkDragRootElement') rootElementSelector: string;
+
+  /** Whether starting to drag this element is disabled. */
+  @Input('cdkDragDisabled')
+  get disabled(): boolean {
+    return this._disabled || (this.dropContainer && this.dropContainer.disabled);
+  }
+  set disabled(value: boolean) {
+    this._disabled = coerceBooleanProperty(value);
+  }
+  private _disabled = false;
+
+  /** Emits when the user starts dragging the item. */
+  @Output('cdkDragStarted') started: EventEmitter<CdkDragStart> = new EventEmitter<CdkDragStart>();
+
+  /** Emits when the user stops dragging an item in the container. */
+  @Output('cdkDragEnded') ended: EventEmitter<CdkDragEnd> = new EventEmitter<CdkDragEnd>();
+
+  /** Emits when the user has moved the item into a new container. */
+  @Output('cdkDragEntered') entered: EventEmitter<CdkDragEnter<any>> =
+      new EventEmitter<CdkDragEnter<any>>();
+
+  /** Emits when the user removes the item its container by dragging it into another container. */
+  @Output('cdkDragExited') exited: EventEmitter<CdkDragExit<any>> =
+      new EventEmitter<CdkDragExit<any>>();
+
+  /** Emits when the user drops the item inside a container. */
+  @Output('cdkDragDropped') dropped: EventEmitter<CdkDragDrop<any>> =
+      new EventEmitter<CdkDragDrop<any>>();
+
+  /**
+   * Emits as the user is dragging the item. Use with caution,
+   * because this event will fire for every pixel that the user has dragged.
+   */
+  @Output('cdkDragMoved') moved: Observable<CdkDragMove<T>> =
+      Observable.create((observer: Observer<CdkDragMove<T>>) => {
+        const subscription = this._dragRef.moved.pipe(map(movedEvent => ({
+          source: this,
+          pointerPosition: movedEvent.pointerPosition,
+          event: movedEvent.event,
+          delta: movedEvent.delta
+        }))).subscribe(observer);
+
+        return () => {
+          subscription.unsubscribe();
+        };
+      });
+
+  constructor(
+    /** Element that the draggable is attached to. */
+    public element: ElementRef<HTMLElement>,
+    /** Droppable container that the draggable is a part of. */
+    @Inject(CDK_DROP_LIST) @Optional() @SkipSelf()
+    public dropContainer: CdkDropList,
```

Comment:
Could you change this directive to use `DropListRef` internally and have `CdkDropList` provide itself as `DropListRef`? The only place it's referenced directly is to get `disabled`, which also exists on `DropListRef`. The `dropContainer` would still have to be exposed publicly as deprecated, but it could eventually get rid of it. 

Another change I'd like to make for CdkDrag would be to reference the idea of the more generic "drop container" rather than the specific "drop list", where the container is the lower-level thing that just knows about receiving drops and not about lists/reordering.